lib/db/models/SwapDeal.ts

```diff
@@ -0,0 +1,39 @@
+import Sequelize from 'sequelize';
+import { db } from '../../types';
+
+export default (sequelize: Sequelize.Sequelize, DataTypes: Sequelize.DataTypes) => {
+  const attributes: db.SequelizeAttributes<db.SwapDealAttributes> = {
+    role: { type: DataTypes.TINYINT, allowNull: false },
+    state: { type: DataTypes.TINYINT, allowNull: false },
+    phase: { type: DataTypes.TINYINT, allowNull: false },
+    errorReason: { type: DataTypes.STRING, allowNull: true },
+    r_hash: { type: DataTypes.STRING, allowNull: false, unique: true },
+    r_preimage: { type: DataTypes.STRING, allowNull: true },
```

Comment:
Also please make these `rHash` and `rPreimage` since we'll be changing those names internally and it'd be nice not to have to change the schema too.